lib/spack/spack/test/stage.py

```diff
@@ -881,7 +883,6 @@ def test_stage_create_replace_path(tmp_build_stage_dir):
     assert os.path.isdir(stage.path)
```

Comment:
I'm not clear what this test is supposed to be testing. If `name=""`, then `stage.path` will be the Stage root, so this assert will always be true.

If this test wants to assert that the stage is free to overwrite `Stage.path`, then it would need to initialize like `Stage(str(nondir), path=join(test_stage_path, 'afile')` or `Stage(str(nondir), name='afile')`; the assertion at the end should be `assert os.path.isdir(nondir)`, since I think what we're trying to assert is that the Staging logic was willing to remove the original `afile` and replace it with a directory used for managing the stage.

This test was originally added by @tldahlgren in 0ea6e0f8176 so I wanted to check with @tldahlgren: do you agree with the above reasoning?

Reply:
That's the way I read the test too. I added your recommended changes.

Reply:
>... I think what we're trying to assert is that the Staging logic was willing to remove the original `afile` and replace it with a directory used for managing the stage.

I agree that was the goal.